### ISSUE TITLE
Dispatch function not getting called

### ISSUE THREAD

**Issue by umasankar-yedida** [dai-shi/react-tracked#49]
I have tried the Todo useReducer example and did some modifications. 

I have created a wrapper of `input` and throwing `onChange` callback to `TodoList` which will call `dispatch`. As per the expected behaviour, the dispatch function should be called and it should update the filter. But, the dispatch function is not working as expected and the global state is not updating. I'm thinking that it is getting called but for some reason it is not calling the reducer function.

Please check the code @ https://codesandbox.io/s/react-tracked-issue-i5dnq

I have noticed that it is working as expected with HTML's input and material UIs Button. But not working If I create a wrapper or with material UIs MUISelect ( checked with only these so far :) )

**Comment by dai-shi** [comment]
Hi, thanks for giving it a try.

I tried your sandbox and know what's happening.
Basically, you are ignoring the eslint warning: `React Hook React.useCallback has a missing dependency: 'state.listOfTodos'. Either include it or remove the dependency array.`

BTW, I found you seem to have tried with bare contexts, which is good. I did try with it, and the behavior is the same. It only re-renders without any change.

Here's the fixed version: https://codesandbox.io/s/react-tracked-issue-yew3m

(Now, I kinda know this is not an expected answer, because `useCallback` is not effective at all. That's another topic. Let me first respond with this.)
